source/ChiPSeqReader.py

```diff
@@ -362,6 +362,32 @@ def duplicate_region(self, interval):
         self.chr_data[interval.chr].sort_values(by=["start"], inplace=True)
         assert len(self.chr_data[interval.chr]) - debug == old_length
 
+    def duplicate_region_v2(self, interval):                     # Modifies data according to the relationship
+        st, en = self.get_interval(interval, return_ids=True)    # between the "start" and "end" of each peak and
+        #print(st, en)                                           # the "start" and "end" of the duplication interval
+        old_length = len(self.chr_data[interval.chr])
+        drop_indices = []
+        dup_indices = []
+        for i in range(st, en + 1):
+            if (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("start")] < interval.start) and \
+                    (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("end")] > interval.end):
+                drop_indices.append(i)                # Duplication interval is within peak interval. Gets indices of destroyed peaks
+            elif (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("start")] > interval.start) and \
+                    (self.chr_data[interval.chr].iloc[i, self.chr_data[interval.chr].columns.get_loc("end")] < interval.end):
+                dup_indices.append(i)                 # Peak interval is within duplication interval. Gets indices of duplicated peaks
+        debug = len(dup_indices) - len(drop_indices)
+        dup_data = self.chr_data[interval.chr].loc[self.chr_data[interval.chr].index[dup_indices]]  # Duplicated peaks as df
```

Comment:
Please provide a test module showing this does work. 
.index should return row labels only, whereas next line attempts to use ["start"] field, which if I understand right will not exist.

Reply:
I can't attach python file here, so there is link where you can download Test_duplication.rar including test file, data file and file with orientations I used.

[https://github.com/vma2-3/3DPredictor-study/blob/d4c8e6aec9545ae5b8ad22c1e2461faf8e1231f3/Test_duplication.rar](url)

Reply:
After using .index we receive labels of rows to be duplicated and create a new df "dup_data" from rows with corresponding labels (line 379). "dup_data" contains all columns as the original df.